solr/core/src/java/org/apache/solr/search/CollapsingQParserPlugin.java

```diff
@@ -128,6 +128,28 @@ field collapsing (with ngroups) when the number of distinct groups
   public static final String HINT_TOP_FC = "top_fc";
   public static final String HINT_MULTI_DOCVALUES = "multi_docvalues";
 
+  public enum NullPolicy {
+    IGNORE("ignore", 0),
```

Comment:
Can we get rid of the earlier String declarations too? Lines 125-127?

Reply:
@madrob Removed them. Added an unit test to validate if null policy is an illegal argument as well.

Reply:
I agree with @madrob about removing the unused declarations but as access the public for these variables. I think we should deprecate it in this PR(for 8x) and remove it in different PR
wdyt?